initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/scm/git/GitProjectGenerationConfiguration.java

```diff
@@ -72,6 +72,9 @@ private GitIgnore createGitIgnore() {
 		gitIgnore.getIntellijIdea().add(".idea", "*.iws", "*.iml", "*.ipr");
 		gitIgnore.getNetBeans().add("/nbproject/private/", "/nbbuild/", "/dist/",
 				"/nbdist/", "/.nb-gradle/");
+		gitIgnore.getVisualStudioCode().add("vscode/*", "!.vscode/settings.json",
```

Comment:
While vendors may provide a different recommendation, our strategy is to exclude all IDE related files from source control. I'd change this to exclude the whole `.vscode` directory